node/src/components/block_proposer.rs

```diff
@@ -364,263 +385,62 @@ impl BlockProposerReady {
     }
 
     /// Returns a list of candidates for inclusion into a block.
-    // TODO:
-    /// rename to proposed deploys
-    // TODO:
-    /// maybe use cuckoofilter
-    fn propose_deploys(
+    fn propose_proto_block(
         &mut self,
         deploy_config: DeployConfig,
         block_timestamp: Timestamp,
-        past_deploys: HashSet<DeployHash>,
-    ) -> HashSet<DeployHash> {
-        // deploys_to_return = all deploys in pending that aren't in finalized blocks or
+        past_deploys: &[DeployHash],
```

Comment:
Wouldn't it be better to keep this as a set, since we only use it for membership checks?